### ISSUE TITLE
The protected page is not working as expected.

### ISSUE THREAD

**Issue by hmzshin** [hmzshin/ecommerce#34]
I've created a protected route with React Router that checks if the user is logged in before accessing the order page. Using the useEffect hook, I make a fetch request to an external API to retrieve the user's saved addresses, but only when the page is loaded.

I encountered an issue with this scenario: when a user, who isn't logged in, adds products to the cart and then attempts to access the order page, they're directed to the login page. After logging in with their email and password, the saved addresses fail to load. However, upon refreshing the page, the saved addresses load correctly.

I save the token upon successful login, but the fetch request for addresses isn't created with the token, resulting in an unauthorized request and the failure to fetch addresses.

You can find the code for my project [here](https://github.com/hmzshin/ecommerce).

@gokhanozdemir @erhanfirat

**Comment by hmzshin** [comment]
I'm sharing a short video that demonstrates the problem I've encountered.

https://github.com/hmzshin/ecommerce/assets/59126754/72ee769f-81ab-41be-9fdf-4b08eb78ce65

**Comment by gokhanozdemir** [comment]
Could you please try connecting your useEffect in App.tsx line 42, to a boolean variable connected to the user held in store?  It should be boolean so that it will not re-render if token is updated. 

const isLoggedIn = user.name.length > 0 ?  true : false


```
useEffect(() => {
    Promise.all([
      dispatch(fetchRoles()),
      dispatch(fetchCategories()),
      verifyUser(),
    ]);
  }, [isLoggedIn]);
```

**Comment by hmzshin** [comment]
Normally, I make Axios requests by checking if there is a token in the local storage. However, when I redirected to the OrderPage, the fetching occurred before the token was written to local storage. Consequently, the fetch request happened without the token, causing an issue.

To resolve this, I saved the token information to the Redux state and modified the fetch request to use the token stored within the state. Below is the code snippet I added to address this issue:

```useEffect(() => {
    if (userToken) {
      axios
        .get("https://workintech-fe-ecommerce.onrender.com/user/address", {
          headers: {
            Authorization: userToken,
          },
        })
        .then((res) => dispatch(setUserAddresses(res.data)))
        .catch((error) => console.log(error));
    }
  }, [userToken]);
```  
Thanks to @gokhanozdemir for his help in resolving this issue.
